### PR TITLE
Updating custom footer for GCWeb App templates

### PR DESCRIPTION
**Pull request by ahmad-shahid**

### CODE REVIEW

**Comment by ahmad-shahid** [comment]
@plup2 Hello, regarding the GCIntranet custom footer, I didn't make any changes. There is code already for backwards compatibility.

Also, it is different than GCWeb because it has sections, rather than just links. I think we should go over it as part of a separate issue as there will be a few things to discuss.

File: src/gcweb/appFooter.ejs [hunk 1]
```diff
@@ -6,55 +6,56 @@
...
-                <ul class="list-unstyled colcount-sm-2 colcount-md-3">
-                    <%_ for (var sectionIndex=0; sectionIndex<pr.footerSections.length; sectionIndex++) { 
-                        const link=pr.footerSections[sectionIndex];-%>
-                        <li><a href="<%- link.href %>"<%- (link.newWindow==true)? ' target="_blank"  rel="noopener"': '' %>><%= link.text %><%_ if (link.newWindow==true) { -%><span class="wb-inv"> <^- msg('This link will open in a new window') ^></span><%_ } -%></a></li>
-                    <%_ } -%>
-                </ul>
-            </nav>
+    <%_ if (pr.footerSections != null || (pr.contactLink != null && pr.contactLink.length > 0)) { -%>
+        <%# Check if footerSections is an array to maintain backwards compatibility -%>
+        <%_ if (pr.footerSections != null) { -%>
```
Comment (plup2) [inline src/gcweb/appFooter.ejs:11]: This should be just a single `if`

File: src/gcweb/appFooter.ejs [hunk 2]
```diff
@@ -6,55 +6,56 @@
...
+            <%_ if (Array.isArray(pr.footerSections)) pr.footerSections = { links: pr.footerSections } -%>
+        <%_ } -%>
+        <div class="gc-contextual">
+            <div class="container">
+                <nav>
+                    <h2 class="wb-inv"><^- msg('Related links') ^></h2>
+                    <%_ if (pr.footerSections.title != null) { -%> <h3><%= pr.footerSections.title %></h3> <%_ } -%>
```
Comment (plup2) [inline src/gcweb/appFooter.ejs:18]: Should check that `pr.footerSections != null && pr.footerSections.title != null`

(and add a test for the "contact link but no footer section situation)

File: src/gcweb/appFooter.ejs [hunk 3]
```diff
@@ -6,55 +6,56 @@
...
-                <div class="col-xs-6 col-md-3 col-lg-2 text-right">
-                    <img src="<%- wet.builder.environment(pr.cdnEnv) %>wet-boew/assets/wmms-blk.svg" alt="<^- msg('Symbol of the Government of Canada') ^>">
-                </div>
+                </ul>
+            </nav>
+            <div class="col-xs-6 visible-sm visible-xs tofpg">
+                <a href="#wb-cont"><^- msg('Top of Page') ^> <span class="glyphicon glyphicon-chevron-up"></span></a>
+            </div>
+            <div class="wtrmrk align-self-end">
+                <img src="https://wet-boew.github.io/themes-dist/GCWeb/GCWeb/assets/wmms-blk.svg" alt="<^- msg('Symbol of the Government of Canada') ^>">
```
Comment (plup2) [inline src/gcweb/appFooter.ejs:58]: (*also applies to footer, and wherever else links like this one could have slipped through*)
Link should be to self-hosted image instead.

**Review by plup2 — CHANGES_REQUESTED** [review]

File: src/gcweb/appFooter.ejs (hunk 1 above)
Comment (ahmad-shahid) [reply src/gcweb/appFooter.ejs]: done

File: src/gcweb/appFooter.ejs (hunk 2 above)
Comment (ahmad-shahid) [reply src/gcweb/appFooter.ejs]: done

File: src/gcweb/appFooter.ejs (hunk 3 above)
Comment (ahmad-shahid) [reply src/gcweb/appFooter.ejs]: done

**Review by plup2 — APPROVED** [review]

**Pull request merged**